dlx_rest/static/js/file_search.js

```diff
@@ -48,172 +48,187 @@ async function fetchResults(text, option, url) {
-  }
+    thead.appendChild(tr);
 
-  fileObjectArray.forEach((element) => {
-    const row = document.createElement("tr");
+    let th_txt = ["File Name", "Document Symbol", "Language(s)", "Action", "Link"];
 
-    //column #1 - name of each file
-    const col_1 = document.createElement("td");
-    col_1.textContent = element.name;
-
-    //column #2 - document symbol
-    const col_2 = document.createElement("td");
-    col_2.textContent = element.docSymbol;
-    col_2.setAttribute("contenteditable", true); //new
-
-    col_2.addEventListener("blur", function (e) {
-      element.updateSymbol(this.textContent);
-
-    });
-
-    //column #3 - language
-    const col_3 = document.createElement("td");
-    const lang_div = document.createElement("div");
-    //English
-    const span_e = languagePill(element, "EN", "outlined-primary", "bg-primary");
-    lang_div.appendChild(span_e);
-
-    //French
-    const span_f = languagePill(element, "FR", "outlined-secondary", "bg-secondary");
-    lang_div.appendChild(span_f);
-
-    //Spanish
-    const span_s = languagePill(element, "ES", "outlined-success", "bg-success");
-    lang_div.appendChild(span_s);
-
-    //Arabic
-    const span_a = languagePill(element, "AR", "outlined-warning", "bg-warning");
-    lang_div.appendChild(span_a);
-
-    //Chinese
-    const span_c = languagePill(element, "ZH", "outlined-danger", "bg-danger");
-    lang_div.appendChild(span_c);
-
-    //Russian
-    const span_r = languagePill(element, "RU", "outlined-dark", "bg-dark");
-    lang_div.appendChild(span_r);
-
-    //German
-    const span_g = languagePill(element, "DE", "outlined-info", "bg-info");
-    lang_div.appendChild(span_g);
-
-    col_3.appendChild(lang_div);
-
-    for (let l in element.language) {
-      switch (element.language[l]) {
-        case "EN":
-          span_e.classList.replace("outlined-primary", "bg-primary");
-          break;
-        case "FR":
-          span_f.classList.replace("outlined-secondary", "bg-secondary");
-          break;
-        case "ES":
-          span_s.classList.replace("outlined-success", "bg-success");
-          break;
-        case "AR":
-          span_a.classList.replace("outlined-warning", "bg-warning");
-          break;
-        case "ZH":
-          span_c.classList.replace("outlined-danger", "bg-danger");
-          break;
-        case "RU":
-          span_r.classList.replace("outlined-dark", "bg-dark");
-          break;
-        case "DE":
-          span_g.classList.replace("outlined-info", "bg-info");
-          break;
-        default:
-          break;
-      }
+    for (let t in th_txt) {
+        const th = document.createElement("th");
+        th.textContent = th_txt[t];
+        tr.appendChild(th);
     }
 
+    for (const file of results) {
+        fileObjectArray.push(new FileContent(file._id, file.filename, file.docsymbol, file.languages, file.uri));
```

Comment:
We need to make sure here that we're using the web application to resolve filenames, e.g., instead of 

https://undl-files.s3.amazonaws.com/69d8ae97159b0fb6d2c48925ebcce2de (a production URL)

We resolve to:

https://{site url}/api/files/69d8ae97159b0fb6d2c48925ebcce2de?action=open (a URL that depends on the deployed environment, e.g., dev or qat)

This can be accomplished by modifying the data in routes.py to point to an api_file object instead of the direct URI to the file.

Reply:
Are you asking me to call the API to get the file when the url of the file is already available in the set of data returned by the backend?
If it's the case I think this kind of approach should be part of the requirements and be identified as an internal rule.
The way the link is created should work everywhere because even if the environment is changing the location of the file is still the same on S3 , am I wrong?